setup.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+"""
+setup.py implementation, interesting because it parsed the first __init__.py and
+    extracts the `__author__` and `__version__`
+"""
+
+import sys
+from ast import Assign, Name, parse
+from functools import partial
+from itertools import chain
+from operator import attrgetter
+from os import listdir, path
+from os.path import extsep
+
+from setuptools import find_packages, setup
+
+if sys.version_info[:2] >= (3, 12):
+    from ast import Del as Str
```

Comment:
This seems a bit hacky, is this needed ?

Reply:
Removed Python 2 support